src/libcore/ptr.rs

```diff
@@ -10,9 +10,49 @@
 
 // FIXME: talk about offset, copy_memory, copy_nonoverlapping_memory
 
-//! Raw, unsafe pointers, `*const T`, and `*mut T`.
+//! Manually manage memory through raw pointers.
 //!
 //! *[See also the pointer primitive types](../../std/primitive.pointer.html).*
+//!
+//! # Safety
+//!
+//! Many functions in this module take raw pointers as arguments and dereference
+//! them. For this to be safe, these pointers must be valid. However, because
+//! rust does not yet have a formal memory model, determining whether an
+//! arbitrary pointer is valid for a given operation can be tricky.
+//!
+//! There are two types of operations on memory, reads and writes. It is
+//! possible for a `*mut` to be valid for one operation and not the other. Since
+//! a `*const` can only be read and not written, it has no such ambiguity. For
```

Comment:
This makes the types mean much more than they should, I think. One can cast freely and in safe code between `*mut` and `*const`. For all intents and purposes, these are the same type; `*const` just has a "lint" associated with it that you cannot write through it.